server/src/main/java/org/opensearch/cluster/ExtensionRequest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.extensions.ExtensionsOrchestrator;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * CLusterService Request for Extensibility
+ *
+ * @opensearch.internal
+ */
+public class ExtensionRequest extends TransportRequest {
+    private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
+    private ExtensionsOrchestrator.RequestType requestType;
+
+    public ExtensionRequest(ExtensionsOrchestrator.RequestType requestType) {
+        this.requestType = requestType;
+    }
+
+    public ExtensionRequest(StreamInput in) throws IOException {
+        super(in);
+        this.requestType = in.readEnum(ExtensionsOrchestrator.RequestType.class);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeEnum(requestType);
+    }
+
+    public ExtensionsOrchestrator.RequestType getRequestType() {
+        return this.requestType;
+    }
+
+    public String toString() {
+        return "ClusterServiceRequest{" + "requestType=" + requestType + '}';
```

Comment:
ExtensionRequest is the main generic extension point request sent to extensions. The toString override should print out "ExtensionRequest{requestType=...}" instead of "ClusterServiceRequest{requestType=...}'